src/MigrationTools.Clients.AzureDevops.Rest/Processors/AzureDevOpsPipelineProcessor.cs

```diff
@@ -591,11 +629,35 @@ private async Task<IEnumerable<Mapping>> CreateServiceConnectionsAsync()
 
             var sourceDefinitions = await Source.GetApiDefinitionsAsync<ServiceConnection>();
             var targetDefinitions = await Target.GetApiDefinitionsAsync<ServiceConnection>();
-            var mappings = await Target.CreateApiDefinitionsAsync(FilterOutExistingDefinitions(sourceDefinitions, targetDefinitions));
-            mappings.AddRange(FindExistingMappings(sourceDefinitions, targetDefinitions, mappings));
+
+            var updatedSourceDefinitions = UpdateProjectReferences(sourceDefinitions, targetDefinitions, await Source.GetProject(), await Target.GetProject());
+            var mappings = await Target.CreateApiDefinitionsAsync(FilterOutExistingDefinitions(updatedSourceDefinitions, targetDefinitions));
+            mappings.AddRange(FindExistingMappings(updatedSourceDefinitions, targetDefinitions, mappings));
             return mappings;
         }
 
+        private IEnumerable<ServiceConnection> UpdateProjectReferences(IEnumerable<ServiceConnection> sourceDefinitions, IEnumerable<ServiceConnection> targetDefinitions, Project sourceProject, Project targetProject)
+        {
+            Log.LogInformation($"Updating Service Connections project references..");
+
+            var projectReference = targetDefinitions.First().ServiceEndpointProjectReferences.First().ProjectReference;
```

Comment:
This is not safe since you don't know for sure that the target project has any existing service connections. Just by executing an basic test where I wanted to migrate a single build pipeline that uses a service connection to another project, that didn't have anything, this line threw an error.

Reply:
This variable also isn't used anywhere. Couldn't it be just removed?